Add TEAM4 mesh refinement test case

EdwardPalmer99 opened the pull request:
**Changes**
- Added Reset method to FluxMonitorAux
- Added a second test case "TestTEAM4HFormMeshUpdates" which is identical to "TestTEAM4HForm" but solves for several different refinement levels.

Commit:
Added additional mesh update test case to test_team4_Hform which runs problem for different refinement levels.

Commit:
Added Reset method to FluxMonitorAux.

alexanderianblair commented on test/integration/test_team4_hform.cpp:
```diff
@@ -182,3 +182,112 @@ TEST_CASE_METHOD(TestTEAM4HForm, "TestTEAM4HForm", "[CheckRun]")
+
+  problem_builder->FinalizeProblem();
+
+  auto problem = problem_builder->ReturnProblem();
```
Could put the problem setup in a method and share it with the existing TEAM 4 test? That way we won't need to change it in two places if/when we make changes in future, and it should make these tests more readable

Commit:
Fixes issue with test4 where the serial mesh was mistakenly refined rather than the ParMesh.

Commit:
Rebuild a0 solver in ScalarPotentialSource when Apply called.

Required for updates to work correctly. Refer to TestTEAM4HFormMeshUpdates test.

Commit:
Merge branch 'EdwardPalmer99/mesh-update-feature/rebuild-scalar-potential-source-a0-solver' into EdwardPalmer99/mesh-update-feature/general-case-bcs-update

Commit:
Reduces max refinement level due to time taken to execute problem.

Commit:
Adds "[!benchmark]" tag to TestTeam4HFormMeshUpdates to ensure it is skipped when running test cases because it is a very heavy test.

Commit:
Adds "ConfigureProblem" method to TestTEAM4HForm to reduce code duplication.

Commit:
Adds DefineExecutionerParameters method.

Commit:
Adds ExtractPeakCurrentAndTime method.

Commit:
Adds "VerifyPeakCurrentAndTimeWithinTolerances".

Commit:
Updates comments in TestTEAM4HForm.

Commit:
Moves pmesh refinement to top of for loop.

Commit:
Removes use of unique pointer for TransientExecutioner in TestTEAM4HForm.
